x-pack/platform/plugins/shared/lens/public/app_plugin/shared/edit_on_the_fly/lens_configuration_flyout.tsx

```diff
@@ -104,8 +105,10 @@ export function LensEditConfigurationFlyout({
             return customIsEqual(
               previousAttrs.state.visualization,
               previousAttrs.references,
+              previousAttrs.state.datasourceStates.formBased,
               visualizationState,
               attributes.references,
+              datasourceStates?.formBased?.state as FormBasedPersistedState,
```

Comment:
```suggestion
              datasourceStates,
```

Reply:
This was another reason I just passed `formBased` because this instance is not the state directly but `DatasourceState` which adds the `isLoading` property and nests the actual `state`. This would require having a union parameter to accept both cases and resolve later.

```ts
export interface DatasourceState {
  isLoading: boolean;
  state: unknown;
}
```

Reply:
Was handled the nested version by abstracting to a util function that effectively does `datasourceState.state ?? datasourceState`.

Fixed in 878a48556a848b8d34945cb7d90713e1a254928b